apps/api_web/test/api_web/controllers/route_pattern_controller_test.exs

```diff
@@ -479,6 +479,203 @@ defmodule ApiWeb.RoutePatternControllerTest do
 
       assert [%{"sort_order" => 1}, %{"sort_order" => 2}, %{"sort_order" => 3}] = attrs
     end
+
+    test "can filter by date", %{conn: base_conn} do
+      today = Parse.Time.service_date()
+      bad_date = ~D[2017-01-01]
+
+      service = %Model.Service{
+        id: "service",
+        start_date: today,
+        end_date: today,
+        added_dates: [today]
+      }
+
+      route = %Model.Route{id: "route"}
+      route_pattern = %Model.RoutePattern{id: "rp1", route_id: "route"}
+
+      trip = %Model.Trip{
+        id: "trip",
+        route_id: route.id,
+        service_id: service.id,
+        route_pattern_id: route_pattern.id
+      }
+
+      State.Service.new_state([service])
+      State.Route.new_state([route])
+      State.RoutePattern.new_state([route_pattern])
+      State.Trip.new_state([trip])
+      State.Trip.reset_gather()
+      State.RoutesByService.update!()
+
+      today_iso = Date.to_iso8601(today)
+      bad_date_iso = Date.to_iso8601(bad_date)
+
+      params = %{"filter" => %{"date" => today_iso}}
+      data = ApiWeb.RoutePatternController.index_data(base_conn, params)
+      assert [route_pattern] == data
+
+      params = put_in(params["filter"]["date"], bad_date_iso)
+      data = ApiWeb.RoutePatternController.index_data(base_conn, params)
+      assert [] == data
+
+      params = %{"filter" => %{"date" => ""}}
+      data = ApiWeb.RoutePatternController.index_data(base_conn, params)
+      assert [route_pattern] == data
+
+      params = %{"filter" => %{"date" => "invalid"}}
+      data = ApiWeb.RoutePatternController.index_data(base_conn, params)
+      assert [route_pattern] == data
+    end
+
+    test "can filter by date and route(s)", %{conn: base_conn} do
+      today = Parse.Time.service_date()
+      bad_date = ~D[2027-01-01]
```

Comment:
Will this cause problems when today actually is 1/1/27? You could define this to be today's date plus a year instead.